### PR TITLE
Knowledge transfer write-up for automate maintenance

### PR DESCRIPTION
**Pull request by msorens**
# THIS PR is about writing up the process for maintaining Automate -- see maintenance.md.

# The text presented just below is tech debt that surfaces when following that write-up; each item here should migrate to its own issue.

_UPDATE: All items below now have issues in epic #4696 (bringing dependencies up-to-date)_

Following the process in the attached document, the following issues need to be addressed promptly:

1. Netlify: "Deploy did not complete for chef-automate" -- this message is being generated for every PR for some weeks now.
image

2. Version of node.js needs to be upgraded from 12 to 14 per the guidelines on their site (https://nodejs.org/en/about/releases/).

3. About 2 dozen npm packages need to be updated for automate-ui -- run `npm outdated` from components/automate-ui. 

4. TSLint support discontinued -- from components/automate-ui run `make lint` and this warning appears:
```
TSLint's support is discontinued and we're deprecating its support in Angular CLI.
To opt-in using the community driven ESLint builder,
see: https://github.com/angular-eslint/angular-eslint#migrating-from-codelyzer-and-tslint.
```

5. karma-coverage-istanbul-reporter is deprecated -- from components/automate-ui run `make unit` and this warning appears:
```
'karma-coverage-istanbul-reporter' usage has been deprecated since version 11.
Please install 'karma-coverage' and update 'karma.conf.js.' 
For more info, see https://github.com/karma-runner/karma-coverage/blob/master/README.md
```

6. protractor error -- from components/automate-ui run `make e2e` and this error appears:
```
spawn Unknown system error -86
References:
  https://stackoverflow.com/questions/65618558/osx-fix-selenium-chromedriver-launch-error-spawn-unknown-system-error-86-bad-cp
  https://github.com/angular/webdriver-manager/issues/476
```

### :nut_and_bolt: Description: What code changed, and why?
Adding maintenance.md to describe how to maintain Automate.
I suggest pressing the "Rich Diff" button, as indicated, when reviewing the markdown:
image

### :chains: Related Resources
NA

### :+1: Definition of Done
All engineers have clear and complete information so they can contribute to maintaining the Automate code base.

### :athletic_shoe: How to Build and Test the Change
Review the document.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [x] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: dev-docs/maintenance.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+- https://github.com/golang/go/releases
+- https://github.com/grpc/grpc-go/releases
+- https://github.com/golang/protobuf/releases
+- https://github.com/open-policy-agent/opa/releases
+- https://github.com/grpc-ecosystem/grpc-gateway/releases
+- https://github.com/golang/protobuf/protoc-gen-go/releases
+- https://github.com/stretchr/testify/releases
+- https://github.com/lib/pq/releases
+- core/dex
+- core/nginx
```
Comment (danielsdeleo) [inline dev-docs/maintenance.md:32]: Why are these listed as hab core packages vs. the other ones that have GitHub URLs here?

**Review by danielsdeleo — APPROVED** [review]
This can be merged as-is if you like, as it's much better than not having it. 

That said, I'd consider making this two files with more verbose file names. "maintenance" is kinda generic and the the dependency updates and pipeline things are pretty different tasks. A file name like `how_to_detect_and_fix_pipeline_issues.md` is a bit more enticing in a file name list.

File: dev-docs/maintenance.md (hunk 1 above)
Comment (msorens) [reply dev-docs/maintenance.md:32]: Only because I did not know immediately where to find those references. 🤷

**Review by jaym — APPROVED** [review]

File: dev-docs/maintenance.md (hunk 1 above)
Comment (msorens) [reply dev-docs/maintenance.md]: Fixed.

**Comment by msorens** [comment]
@danielsdeleo Spurred on by your comment, I found there was already a pipeline-related doc, so I moved my pipeline-monitoring section there (HOW_THE_PIPELINE_WORKS.md) and then renamed the dependency portion (HOW_TO_MAINTAIN_A2.md). I was tempted to call it HOW_TO_AVOID_CRASH_AND_BURN.md but, well...

**Pull request merged**